### ISSUE TITLE
Wrong type for return and parameters

### ISSUE THREAD

**Issue by InkosiZhong** [mitroadmaps/gomapinfer#2]
Hi, I recently loaded your package in another project and I get some errors when I run it.
I'm new to golang. I suspect that rtreego has related updates that modified the type of return and parameters, and this project is not adapted.
I'm wondering if there is an easy fix? Or maybe I can rolling back to a certain version of rtreego? (which version?)
Thank you so much.

> Environment:
> Ubuntu 20.04
> go1.19.1

```bash
# github.com/mitroadmaps/gomapinfer/common
../go/src/github.com/mitroadmaps/gomapinfer/common/graph_rtree.go:16:9: cannot use rect (variable of type rtreego.Rect) as type *rtreego.Rect in return statement
../go/src/github.com/mitroadmaps/gomapinfer/common/graph_rtree.go:38:41: cannot use RtreegoRect(rect) (value of type *rtreego.Rect) as type rtreego.Rect in argument to rtree.tree.SearchIntersect
../go/src/github.com/mitroadmaps/gomapinfer/common/graph_rtree.go:41:14: impossible type assertion: spatials[i].(*edgeSpatial)
	*edgeSpatial does not implement rtreego.Spatial (wrong type for Bounds method)
		have Bounds() *rtreego.Rect
		want Bounds() rtreego.Rect
../go/src/github.com/mitroadmaps/gomapinfer/common/graph_rtree.go:49:16: cannot use &edgeSpatial{…} (value of type *edgeSpatial) as type rtreego.Spatial in argument to rtree.Insert:
	*edgeSpatial does not implement rtreego.Spatial (wrong type for Bounds method)
		have Bounds() *rtreego.Rect
		want Bounds() rtreego.Rect
```

**Comment by imyumichan** [comment]
I faced the same issue. I found that the problem is caused by a recent change in rtreego (https://github.com/dhconnelly/rtreego/commit/2008b37524910b3e54046f1f7b12139316a0461d). Rect is changed from a pointer to a struct object. I did a similar change here #3 and it works for me now.

**Comment by imyumichan** [comment]
Hey, I restored the checking on empty struct (in a right way). You can check my new commit if needed.

This is my first time to use golang. Please correct me if anything is wrong.

**Comment by codeastra2** [comment]
Thanks for the fix @imyumichan! I faced the following issue though: 

```


ramd_rm@othrys5:~/workspace/roadtracer/dataset$ go get github.com/imyumichan/gomapinfer/googlemaps
# github.com/mitroadmaps/gomapinfer/common
../../../go/src/github.com/mitroadmaps/gomapinfer/common/graph_rtree.go:16:2: cannot use rect (type rtreego.Rect) as type *rtreego.Rect in return argument
../../../go/src/github.com/mitroadmaps/gomapinfer/common/graph_rtree.go:38:52: cannot use RtreegoRect(rect) (type *rtreego.Rect) as type rtreego.Rect in argument to rtree.tree.SearchIntersect
../../../go/src/github.com/mitroadmaps/gomapinfer/common/graph_rtree.go:41:25: impossible type assertion:
	*edgeSpatial does not implement rtreego.Spatial (wrong type for Bounds method)
		have Bounds() *rtreego.Rect
		want Bounds() rtreego.Rect
../../../go/src/github.com/mitroadmaps/gomapinfer/common/graph_rtree.go:49:29: cannot use &edgeSpatial literal (type *edgeSpatial) as type rtreego.Spatial in argument to rtree.Insert:
	*edgeSpatial does not implement rtreego.Spatial (wrong type for Bounds method)
		have Bounds() *rtreego.Rect
		want Bounds() rtreego.Rect

```

I still face this error when I tried installing from @imyumichan solution. I believe it is because the import in googlemaps files are still pointing to the old `common/graph_rtree.go` in mitroadmaps. 
Also to use it for downloading the roadtracer dataset I believe this fix needs to be merged.  Please let me know if some workaround exists for downloading the dataset, thanks!

**Comment by codeastra2** [comment]
I  was able to solve it by copying the above mentioned fix onto my local file '../../../go/src/github.com/mitroadmaps/gomapinfer/common/graph_rtree.go' and then executing commands to download the dataset.

**Comment by ty199931** [comment]
> I was able to solve it by copying the above mentioned fix onto my local file '../../../go/src/github.com/mitroadmaps/gomapinfer/common/graph_rtree.go' and then executing commands to download the dataset.

hello, could you send me a file corrected? 1164565117@qq.com
